build_tools/linting.sh

```diff
@@ -13,6 +13,9 @@ echo -e "No problem detected by flake8\n"
 mypy sklearn/
 echo -e "No problem detected by mypy\n"
 
+cython-lint --max-line-length=999 --ignore=E24,E203,E731,E741,W503,W504 .
```

Comment:
I have set a huge limit for the max line length in this PR. It's usually harder to stay below 88 chars in cython and trying to enforce it sometimes makes the code harder to read. I think we can postpone the decision

Reply:
just out of interest, why set a huge limit rather than add `E501` to `--ignore`?

Reply:
I just didn't think about that :)